network/f5/bigip_virtual_server.py

```diff
@@ -0,0 +1,414 @@
+                        except bigsuds.OperationFailed, e:
+                            if "was not found" in str(e):
+                                result['changed']= False
+                            else:
+                                raise
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+
+        elif state == 'present':
+            update = False
+            if not vs_exists(api, name):
+                if (not destination) or (not port):
+                    module.fail_json(msg="both destination and port must be supplied to create a VS")
+                if not module.check_mode:
+                    # a bit of a hack to handle concurrent runs of this module.
+                    # even though we've checked the pool doesn't exist,
+                    # it may exist by the time we run create_pool().
+                    # this catches the exception and does something smart
+                    # about it!
+                    try:
+                        vs_create(api,name,destination,port,pool)
+                        result = {'changed': True}
+                    except bigsuds.OperationFailed, e:
+                        if "already exists" in str(e):
+                            update = True
+                        else:
+                            raise
+                    else:
+                        set_profiles(api,name,all_profiles)
+                        set_snat(api,name,snat)
+                        set_description(api,name,description)
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+            else:
+                update = True
+            if update:
+                # VS exists
+                if not module.check_mode:
+                    # Have a transaction for all the changes
+                    api.System.Session.start_transaction()
+                    result['changed']|=set_destination(api,name,fq_name(partition,destination),port)
+                    result['changed']|=set_pool(api,name,pool)
+                    result['changed']|=set_description(api,name,description)
+                    result['changed']|=set_snat(api,name,snat)
+                    result['changed']|=set_profiles(api,name,all_profiles)
+                    api.System.Session.submit_transaction()
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+
+        elif state in ('disabled', 'enabled'):
+            if name is None:
+                module.fail_json(msg="name parameter required when " \
+                                     "state=enabled/disabled")
+            if not module.check_mode:
+                pass
+            else:
+                # check-mode return value
+                result = {'changed': True}
+
+    except Exception, e:
```

Comment:
we prefer having localized exception handlers so you can give meaningful messages on top of the exception, normally the api libraries give very cryptic responses and it is nicer to give the user something relative to the action being attempted.